File: javascript/rna-transcription/rna-transcription.js
const transcript = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U',
};

const error = () => {
    throw new Error('Invalid input DNA.');
};

export const toRna = dna => {
    return Array.from(dna)
        .map(x => transcript[x] || error())
        .join('');
};

File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(defn ^:private transcript [nucleotide]
  (condp = nucleotide
    \G \C
    \C \G
    \T \A
    \A \U
    (throw (AssertionError. "Invalid nucleotide."))))

(defn to-rna [dna]
  (->> dna
       (map transcript)
       (apply str)))
